### PR TITLE
SPV_EXT_mesh_shader related tools changes

### PR DESCRIPTION
**Pull request by pmistryNV**

### CODE REVIEW

File: source/val/validate_barriers.cpp [hunk 1]
```diff
@@ -45,12 +45,15 @@ spv_result_t BarriersPass(ValidationState_t& _, const Instruction* inst) {
                       model != SpvExecutionModelGLCompute &&
                       model != SpvExecutionModelKernel &&
                       model != SpvExecutionModelTaskNV &&
-                      model != SpvExecutionModelMeshNV) {
+                      model != SpvExecutionModelMeshNV &&
+                      model != SpvExecutionModelTaskEXT &&
+                      model != SpvExecutionModelMeshEXT) {
```
Comment (alan-baker) [inline source/val/validate_barriers.cpp:50]: SPV_EXT_mesh_shader requires SPIR-V 1.4. This check only happens if the binary version is less than 1.3.

Comment (alan-baker) [reply source/val/validate_barriers.cpp:50]: The validation for the 1.4 requirement appears to be missing. Please add it to validate_extensions.cpp.

**Review by alan-baker — CHANGES_REQUESTED** [review]
Please follow up on the spec bug separately, but I'd like to address the version requirement before merging this PR.

File: source/val/validate_atomics.cpp [hunk 2]
```diff
@@ -40,6 +40,7 @@ bool IsStorageClassAllowedByUniversalRules(uint32_t storage_class) {
     case SpvStorageClassImage:
     case SpvStorageClassFunction:
     case SpvStorageClassPhysicalStorageBuffer:
+    case SpvStorageClassTaskPayloadWorkgroupEXT:
```
[the remark this answers is not in the document]
Comment (pmistryNV) [reply source/val/validate_atomics.cpp:43]: Thanks for pointing it out, I will get it updated

File: source/val/validate_barriers.cpp (hunk 1 above)
Comment (pmistryNV) [reply source/val/validate_barriers.cpp]: regarding 1.4 requirement added code to validate_extensions.cpp::ValidateExtension
and removed changes from validate_barriers.cpp

**Comment by unit-a-user** [comment]
Needs new review?

**Review by alan-baker — APPROVED** [review]

**Comment by unit-a-user** [comment]
Test done

**Comment by unit-a-user** [comment]
Finished?

File: source/val/validate_atomics.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (pmistryNV) [reply source/val/validate_atomics.cpp:43]: https://github.com/KhronosGroup/SPIRV-Registry/pull/166 to update the spec

**Pull request merged**